x-pack/filebeat/input/awss3/clean_store.go

```diff
@@ -38,13 +42,17 @@ func gcStore(logger *logp.Logger, started time.Time, store *statestore.Store, st
 	if err := gcClean(store, states, keys); err != nil {
 		logger.Errorf("Failed to remove all entries from the registry: %+v", err)
 	}
+
+	if err := store.Set(awsS3WriteCommitStateKey, commitWriteState{time.Now()}); err != nil {
```

Comment:
this should not be `time.Now()` (dependant on the machine running Filebeat), it should be extracted from the listed objects (`lastModified` field), keeping the newest date. Trusting the local machine to be in sync to S3 dates can lead to sync problems, where the local time may be wrong, or timezones differ.

Reply:
good catch!